File: Easy/344. Reverse String/Solution.cpp
class Solution {
public:
    void reverseString(vector<char>& s) {
    int n = s.size();
    int x = 0;
    int y = n - 1;
    while(x < y){
        std::swap(s[x], s[y]);
        x++;
        y--;
    }
    }
};

File: Easy/344. Reverse String/solution.py
class Solution(object):
    def reverseString(self, s):
        """
        :type s: str
        :rtype: str
        """
        i, j = 0, len(s) - 1
        sList = list(s) # convert str into list, easy to operate
        while i < j:
            tmp = sList[i]
            sList[i] = sList[j]
            sList[j] = tmp
            i += 1
            j -= 1
        return ''.join(sList)